### PR TITLE
chore: verify yarn lock

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

This adds PR verification that `yarn.lock` does not drift from package.json declarations.

##### Successful execution

image

##### Failure injection

https://github.com/aws-amplify/amplify-cli/pull/11941/commits/f7bf33f43b1763ab3aebdfa916a80e91b5bcf5a7

https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/15659/workflows/da24d385-ca9c-4e51-9bc3-ead12ab03c6a

image

image

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/verify-extract-api.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+changed_api_extracts=$(git status | grep -F API.md | wc -l)
+
+if [[ changed_api_extracts -gt 0 ]]; then
+  echo "Fail! Detected api change. Please run 'yarn extract-api' and add API.md file changes to the change set."
+  exit 1;
+else
+  echo "Success! No drift detected in API.md files."
+fi
```
Comment (sobolk) [inline scripts/verify-extract-api.sh:10]: changes in this file.
- changed name to match `-` we use in other script names.
- added `-F` flag to make sure that grep doesn't interpret `.` in `API.md` as wildcard
- added else case to print successful message.

File: .circleci/config.base.yml [hunk 2]
```diff
@@ -176,6 +176,18 @@ jobs:
           command: |
             yarn verify-api-extract
 
+  verify-yarn-lock:
+    <<: *linux-e2e-executor-large
```
Comment (lazpavel) [inline .circleci/config.base.yml:180]: nit: is a large executor necessary for this?

**Review by goldbez — APPROVED** [review]

**Review by lazpavel — APPROVED** [review]

**Review by sdstolworthy — DISMISSED** [review]

File: .circleci/config.base.yml (hunk 2 above)
Comment (sobolk) [reply .circleci/config.base.yml:180]: it seems so. `yarn install` seems to be quite CPU bound.
image

Comment (sobolk) [reply .circleci/config.base.yml:180]: on the other hand, build goes up to 100%, I'll give it a try to decrease size.

**Review by sdstolworthy — APPROVED** [review]

**Review by goldbez — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**